https://bugs.gentoo.org/332839

From 1359a849055a66eddffe7753485e0f29fb85ef31 Mon Sep 17 00:00:00 2001
From: Eric Blake <eblake@redhat.com>
Date: Fri, 4 Jun 2010 06:07:38 -0600
Subject: [PATCH] Fix build on newer glibc.

POSIX allows, but does not require, that <fcntl.h> implicitly
includes <sys/stat.h>.  Until recent glibc, most systems did
this implicit inclusion, and m4 was mistakenly relying on it.

* src/m4.h (includes): Add <sys/stat.h>.
* THANKS: Update.
Reported by Todd Rinaldo.

Signed-off-by: Eric Blake <eblake@redhat.com>
---
 ChangeLog |    7 +++++++
 THANKS    |    2 ++
 src/m4.h  |    1 +
 3 files changed, 10 insertions(+), 0 deletions(-)

2010-06-04  Eric Blake  <eblake@redhat.com>

	Fix build on newer glibc.
	* src/m4.h (includes): Add <sys/stat.h>.
	* THANKS: Update.
	Reported by Todd Rinaldo.

diff --git a/THANKS b/THANKS
index f43e2aa..6f49e78 100644
--- a/THANKS
+++ b/THANKS
@@ -43,6 +43,7 @@ Frank Schwidom          schwidom@impernet.de
 Gary V. Vaughan         gary@gnu.org
 Geoff Russell           grussell@guest.adelaide.edu.au
 Giovanni Toffetti       toffettg@lu.unisi.ch
+Giuseppe Scrivano       gscrivano@gnu.org
 Greg A. Woods           woods@web.apc.org
 Hal Peterson            hrp@pecan.cray.com
 Hoang Uong              hoang@ornews.intel.com
@@ -114,6 +115,7 @@ Steve Williamson        willy@uinpla.npl.uiuc.edu
 Steven Augart           saugart@yahoo.com
 Thomas Klausner         tk@giga.or.at
 Thorsten Ohl            ohl@physics.harvard.edu
+Todd Rinaldo            toddr@cpanel.net
 Tom G. Christensen      tgc@jupiterrise.com
 Tom McConnell           tmcconne@sedona.intel.com
 Tom Quinn               trq@dionysos.thphys.ox.ac.uk
diff --git a/src/m4.h b/src/m4.h
index ccd9936..70dc7a8 100644
--- a/src/m4.h
+++ b/src/m4.h
@@ -32,6 +32,7 @@
 #include <stdbool.h>
 #include <stdint.h>
 #include <string.h>
+#include <sys/stat.h>
 #include <sys/types.h>
 
 #include "binary-io.h"
-- 
1.7.2

